### PR TITLE
Add Operational Field index page in the GOV.UK Design System

### PR DESCRIPTION
**Pull request by davidgisbey**
## Description

This PR moves the organisation show page to the GOV.UK Design System.

1. Ensures the controller renders the correct layout and view based on the whether the user has the  `Preview design system` permission
2.  Adds the view in the GOV.UK Design System
3. Updates any specs that have broken

## Screenshot

image

image

## Trello card

https://trello.com/c/SoM3MwVh/32-listing-page

⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

Follow [these steps](https://guides.rubyonrails.org/upgrading_ruby_on_rails.html) if you are doing a Rails upgrade.

### CODE REVIEW

File: app/views/admin/operational_fields/index.html.erb [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
+        {
+          text: "Name"
+        },
+      ],
+      rows:
+        @operational_fields.map do |operational_field|
+          [
+            {
+              text: tag.span("Edit field of operation", class: "govuk-visually-hidden") +
+              link_to(operational_field.name, edit_admin_operational_field_path(operational_field), class:"govuk-link")
```
Comment (DilwoarH) [inline app/views/admin/operational_fields/index.html.erb:24]:
```suggestion
              text:  link_to("#{tag.span("Edit field of operation", class: "govuk-visually-hidden")} 
 #{operational_field.name}", edit_admin_operational_field_path(operational_field), class:"govuk-link")
```

Link should include the visually hidden text

**Comment by DilwoarH** [comment]
Looks great - one minor change

File: app/views/admin/operational_fields/index.html.erb (hunk 1 above)
Comment (davidgisbey) [reply app/views/admin/operational_fields/index.html.erb]: ah thanks for the review. Good catch. 

Annoyingly i've had to pass it in as html and sanitise it. If i call sanitise on this ☝️ it's already escaped the span.

**Comment by cbernardojr-gds** [comment]
Looks good @davidgisbey 

Few tweaks (please let me know if you want to discuss any of them):

- The "Add field of operation" button should be green
- Decrease the space between the "Add field of operation" button and the table (should be the default button margin-bottom space: 30px or govuk-!-margin-bottom-6)
- Optional: I would suggest renaming the "Add field of operation" to "Create new field of operation"
- Optional: Move the edit link to another column with an "Edit" link
Example in Content Publisher with the green button and "Create new item":
Screenshot 2023-03-09 at 12 35 31

Prototype reference:
https://gds-wh.herokuapp.com/v2/support/fields/
Password: wh1t3h@11

**Pull request merged**